pkg/controllers/externalsecret/externalsecret_controller_test.go

```diff
@@ -1730,6 +1727,28 @@ var _ = Describe("ExternalSecret controller", Serial, func() {
 		}
 	}
 
+	// Checks that secret annotation has been written based on the all the data for merge keys
+	checkMergeSecretDataHashAnnotation := func(tc *testCase) {
+		const secretVal = "someValue"
+		tc.externalSecret.Spec.Target.CreationPolicy = esv1beta1.CreatePolicyMerge
+
+		// create secret beforehand
+		Expect(k8sClient.Create(context.Background(), &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      ExternalSecretTargetSecretName,
+				Namespace: ExternalSecretNamespace,
+			},
+			Data: map[string][]byte{
+				existingKey: []byte(existingVal),
+			},
+		}, client.FieldOwner(FakeManager))).To(Succeed())
+
+		fakeProvider.WithGetSecret([]byte(secretVal), nil)
+		tc.checkSecret = func(es *esv1beta1.ExternalSecret, secret *v1.Secret) {
+			Expect(secret.Annotations[esv1beta1.AnnotationDataHash]).To(Equal("82a8288e398d5354ab110d2244b7c0b1"))
```

Comment:
Using the hash value directly is hard to maintain and understand. Would you create the hash value using `utils.ObjectHash`?

Reply:
Sure. I will refactor other tests as well 👍

Reply:
Done!